### PR TITLE
[5.1] Convert mod_login to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the administrator login module to service provider.

### Testing Instructions

1. You should see module output in administrator login page

mod_login

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/modules/mod_login/src/Helper/LoginHelper.php [hunk 1]
```diff
@@ -25,14 +26,18 @@
...
      * Get an HTML select list of the available languages.
      *
+     * @param   CMSApplicationInterface  $app  The application
+     *
      * @return  string
+     *
+     * @since   __DEPLOY_VERSION__
      */
-    public static function getLanguageList()
+    public function getLanguages(CMSApplicationInterface $app)
```
Comment (sandewt) [inline administrator/modules/mod_login/src/Helper/LoginHelper.php:40]:
```suggestion
    public function getLanguages(Registry $moduleParams, SiteApplication $app)
```
I wonder why you deviate from the previous converted modules? See, among others eg:
https://github.com/joomla/joomla-cms/blob/fba52f5c7066ff6be85de8b8a46e471379802fc8/modules/mod_articles_archive/src/Helper/ArticlesArchiveHelper.php#L47

[EDIT] ~discussion~ On second thought, not a good suggestion.

File: administrator/modules/mod_login/src/Dispatcher/Dispatcher.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    protected function getLayoutData()
+    {
+        $data = parent::getLayoutData();
+
+        $loginHelper = $this->getHelperFactory()->getHelper('LoginHelper');
+
+        $data['langs']            = $loginHelper->getLanguages($this->getApplication());
+        $data['extraButtons']     = AuthenticationHelper::getLoginButtons('form-login');
+        $data['return']           = $loginHelper->getReturnUriString();
```
Comment (sandewt) [inline administrator/modules/mod_login/src/Dispatcher/Dispatcher.php]: alignment
```suggestion
        $data['langs']        = $loginHelper->getLanguages($this->getApplication());
        $data['extraButtons'] = AuthenticationHelper::getLoginButtons('form-login');
        $data['return']       = $loginHelper->getReturnUriString();
```

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 1ac8a37911c8447a5bb76ceea8709d7d005b14a4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42990">issues.joomla.org/tracker/joomla-cms/42990</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 1ac8a37911c8447a5bb76ceea8709d7d005b14a4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42990">issues.joomla.org/tracker/joomla-cms/42990</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42990">issues.joomla.org/tracker/joomla-cms/42990</a>.</sub>

File: administrator/modules/mod_login/src/Helper/LoginHelper.php (hunk 1 above)
Comment (joomlaweby) [reply administrator/modules/mod_login/src/Helper/LoginHelper.php:40]: So may I mark conversation as resolved?

Comment (sandewt) [reply administrator/modules/mod_login/src/Helper/LoginHelper.php:40]: yes

**Comment by richard67** [comment]
RTC is still valid as the change after that was just small code style change.

**Comment by LadySolveig** [comment]
Thank you @joomlaweby 🚀 and also for testing and review @ceford @viocassel @sandewt

**Pull request merged**